### ISSUE TITLE
<md-input-container> Text area does not shrink when all text is cleared

### ISSUE THREAD

**Issue by cmacdonnacha** [angular/material#8135]
See [codepen](http://codepen.io/cmacdonnacha/pen/bpKvJe) example.

Here's the initial **md-input-container** component with a text area:

      <md-input-container class="md-block">
          <label>Enter some text and hit enter a few times. Then delete all the text.</label>
          <textarea rows="5"></textarea>
      </md-input-container>

![image](https://cloud.githubusercontent.com/assets/9777763/14677784/07181720-070b-11e6-82ba-5a16fb7244dc.png)

I then enter some text and hit enter a few times:

![image](https://cloud.githubusercontent.com/assets/9777763/14677819/2eff5280-070b-11e6-86bf-c738eabacb9b.png)

I've set **rows="5"** on the textarea so it only grows to show those 5 rows.

However, when I clear all the text it doesn't shrink again:
![image](https://cloud.githubusercontent.com/assets/9777763/14677878/6d90053a-070b-11e6-8f5b-f6742edc33c9.png)

**Comment by crisbeto** [comment]
Thanks for the report. This is the "intended" behaviour for the current implementation. We have a PR that will address this issue. [You can keep track of it here](https://github.com/angular/material/pull/7991).

**Comment by cmacdonnacha** [comment]
Thanks @crisbeto. I do like the way it starts at 1 row and expands to the max row though. For example we use this in a chat application where it starts off at 1 row but can expand to 3 rows. Is there a way we can also keep this behavior or make it optional at least? But to shrink back down to 1 row once all the text is cleared. Sort of like slack and other chat apps work:

Start off with 1 row:
![image](https://cloud.githubusercontent.com/assets/9777763/14703212/a2f3b9ce-07a6-11e6-9944-f6076b96da1a.png)

As you type more text it expands to 2 rows:
![image](https://cloud.githubusercontent.com/assets/9777763/14703221/b66c59ca-07a6-11e6-924f-fb121e93b895.png)

And if you clear the text it goes back to 1 row:
![image](https://cloud.githubusercontent.com/assets/9777763/14703237/caab47fc-07a6-11e6-927a-c92e466a9e53.png)

I know that we can't compare other UI frameworks but this would be vital for our project. Otherwise we'd have to put some other hacky workaround in once we update Angular Material.

**Comment by crisbeto** [comment]
It should work like that if you removed the `rows` attribute. Although you should keep in mind that in this case it won't have a max height. If that's not intended, you can use this as a temporary workaround until https://github.com/angular/material/pull/7991 gets merged:

1. Remove the `rows` attribute.
2. Add the following somewhere in your CSS:

```css
textarea.md-input {
  max-height: /* whatever max height you want */;
  overflow: auto;
}
```

**Comment by cmacdonnacha** [comment]
Thanks for that. At the moment after you clear the text it stays at 2 rows, whereas I would prefer if it shrank back to 1 row. So when this is merged and I replace **"rows"** with **max-rows="2"**, will it shrink back to 1 row after clearing the text?

**Comment by cmacdonnacha** [comment]
Ah ok great. Thanks for that and well done on the PR.

**Comment by char0n** [comment]
@crisbeto the css workaround doesn't seems to work any more on v1.1.0-rc4-master-2806ec8. The max-height a overflow is being respected, but it is not possible to scroll inside textarea.

Here is codepen to demonstrate: http://codepen.io/anon/pen/xOKzYo

Tested on Chrome: 51.0.2704.63 m

@crisbeto do you have any idea how to make it work again ? Thanks

**Comment by crisbeto** [comment]
@char0n that's because of a `preventDefault` on the textarea in the old implementation (it's removed in the new one). It's a hack, but you can work around it with a directive that removes the scroll listener. It's not ideal, but hopefully it won't be necessary for very long:

```js
.directive('textarea', function($timeout){
 return {
   restrict: 'E',
   link: function(scope, element){
     $timeout(function(){
        element.off('scroll');
     }, 0, false);
   }
 }
});
```

**Comment by char0n** [comment]
@crisbeto yep, that does the trick. Thanks. I'll follow this issue until PR is merged.

**Comment by cmacdonnacha** [comment]
So will this be in merged into the next release?

**Comment by crisbeto** [comment]
It should be.

**Comment by cmacdonnacha** [comment]
Hey, I just pulled this in 1.1.0-rc.5. It does what it says on the tin but one thing I noticed is that the height sort of jerks a bit. It looks like you're setting an inline css style on the textarea element to set the height and padding. So the height starts off at the default and then you seem to override it which is causing it to move which doesn't look ideal. Any suggestions on how to prevent this? Maybe if you set the default height and padding a bit earlier?
